src/linear_algebra/finsupp.lean

```diff
@@ -58,8 +58,10 @@ open_locale classical big_operators
 
 namespace finsupp
 
-variables {α : Type*} {M : Type*} {N : Type*} {R : Type*}
-variables [semiring R] [add_comm_monoid M] [semimodule R M] [add_comm_monoid N] [semimodule R N]
+variables {α : Type*} {M : Type*} {N : Type*} {R : Type*} (S : Type*)
```

Comment:
This distinguishes the implicit and explicit type variables a bit more cleanly.
```suggestion
variables {α M N R : Type*} (S : Type*)
```

Reply:
I decided it made more sense just to make it explicit in a local `section`